robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowViewTreeObserver.java

```diff
@@ -39,4 +40,24 @@ public void fireOnGlobalLayoutListeners() {
   public List<ViewTreeObserver.OnGlobalLayoutListener> getOnGlobalLayoutListeners() {
     return globalLayoutListeners;
   }
+
+  @Implementation
+  public void addOnPreDrawListener(ViewTreeObserver.OnPreDrawListener listener) {
+    this.preDrawListeners.add(listener);
+  }
+
+  @Implementation
+  public void removeOnPreDrawListener(ViewTreeObserver.OnPreDrawListener listener) {
+    this.preDrawListeners.remove(listener);
+  }
+
+  public void fireOnPreDrawListeners() {
```

Comment:
Why can't we just call ViewTreeObserver.dispatchOnPreDraw() instead of shadowing the real implementation?